test/findex/sqlite_test.dart

```diff
@@ -225,31 +282,17 @@ class SqliteFindex {
 
   static List<UidAndValue> upsertEntries(List<UpsertData> entries) {
     log("upsertEntries: start");
-    List<UidAndValue> rejectedEntries = [];
+    final stmt = db.prepare(
+        'INSERT INTO entry_table (uid, value) VALUES (?, ?) ON CONFLICT (uid)  DO UPDATE SET value = ? WHERE value = ?');
     for (final entry in entries) {
-      List<Uint8List> uids = [entry.uid];
-      final resultSet = db.select(
-        'SELECT value FROM entry_table WHERE uid  = ?',
-        uids,
-      );
-      if (resultSet.length > 1) {
-        throw Exception(
-            "Invalid Entry table, found multiple rows for same uid");
-      }
-
-      if (resultSet.isEmpty) {
-        upsertEntry(entry);
-      } else {
-        log("upsertEntries: exist");
-        Uint8List actualValue = resultSet[0]['value'];
-        if (actualValue == entry.oldValue) {
-          upsertEntry(entry);
-        } else {
-          rejectedEntries.add(UidAndValue(entry.uid, actualValue));
-        }
-      }
+      stmt.execute([
+        entry.uid,
+        entry.newValue,
+        entry.newValue,
+        entry.oldValue,
+      ]);
     }
-    return rejectedEntries;
+    return [];
```

Comment:
We should support `rejectedEntries` here, why removing it?

Reply:
the code contained a bug, since there is no transaction involved here:
previous code:
```
  static List<UidAndValue> upsertEntries(List<UpsertData> entries) {
    log("upsertEntries: start");
    List<UidAndValue> rejectedEntries = [];
    for (final entry in entries) {
      List<Uint8List> uids = [entry.uid];
      final resultSet = db.select(
        'SELECT value FROM entry_table WHERE uid  = ?',
        uids,
      );
      if (resultSet.length > 1) {
        throw Exception(
            "Invalid Entry table, found multiple rows for same uid");
      }

      if (resultSet.isEmpty) {
        upsertEntry(entry);
      } else {
        log("upsertEntries: exist");
        Uint8List actualValue = resultSet[0]['value'];
        if (actualValue == entry.oldValue) {
          upsertEntry(entry);
        } else {
          rejectedEntries.add(UidAndValue(entry.uid, actualValue));
        }
      }
    }
    return rejectedEntries;
  }
```

Reply:
Yes this code is buggy but it doesn't mean we should remove all the `rejectedEntries` stuff :-D

Reply:
I'm working on it :)